packages/fast-components/src/form-associated/index.ts

```diff
@@ -0,0 +1,270 @@
+import { attr, emptyArray, FastElement } from "@microsoft/fast-element";
+import { keyCodeEnter } from "@microsoft/fast-web-utilities";
+
+/**
+ * Disable member ordering to keep property callbacks
+ * grouped with property declaration
+ */
+/* tslint:disable:member-ordering */
+export abstract class FormAssociated<
+    T extends HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement
+> extends FastElement {
+    /**
+     * Must evaluate to true to enable elementInternals.
+     * Feature detects API support and resolve respectively
+     */
+    private static cachedFormAssociated: boolean | null = null;
+    public static get formAssociated(): boolean {
+        return FormAssociated.cachedFormAssociated === null
+            ? (FormAssociated.cachedFormAssociated = "ElementInternals" in window)
+            : FormAssociated.cachedFormAssociated;
+    }

```

Comment:
I'd consider pulling this out of the class instead. Something like this:

```TypeScript
const supportsElementInternals = (function() {
  return "ElementInternals" in window;
})();
```

Put that at the module level and then reference the const value thoughout.